svg/types/scripted/SVGGeometryElement.getTotalLength-04.svg

```diff
@@ -0,0 +1,40 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:h="http://www.w3.org/1999/xhtml">
+  <title>SVGGeometryElement.getTotalLength: 'display' and empty path</title>
+  <h:link rel="help" href="https://svgwg.org/svg2-draft/types.html#__svg__SVGGeometryElement__getTotalLength"/>
+  <h:script src="/resources/testharness.js"/>
+  <h:script src="/resources/testharnessreport.js"/>
+  <path id='pathElement1' description='path with display: initial and an empty path'/>
+  <path id='pathElement2' style='display:none' description='path with display: none and an empty path'/>
+  <rect id='rectElement1' description='rect with display: initial and an empty path'/>
+  <rect id='rectElement2' style='display:none' description='rect with display: none and an empty path'/>
+  <circle id='circleElement1' description='circle with display: initial and an empty path'/>
+  <circle id='circleElement2' style='display:none' description='circle with display: none and an empty path'/>
+  <polygon id='polygonElement1' description='polygon with display: initial and an empty path'/>
+  <polygon id='polygonElement2' style='display:none' description='polygon with display: none and an empty path'/>
+  <polyline id='polylineElement1' description='polyline with display: initial and an empty path'/>
+  <polyline id='polylineElement2' style='display:none' description='polyline with display: none and an empty path'/>
+  <ellipse id='ellipseElement1' description='ellipse with display: initial and an empty path'/>
+  <ellipse id='ellipseElement2' style='display:none' description='ellipse with display: none and an empty path'/>
+  <script>
+[
+  'pathElement1',
+  'pathElement2',
+  'rectElement1',
+  'rectElement2',
+  'circleElement1',
+  'circleElement2',
+  'polygonElement1',
+  'polygonElement2',
+  'polylineElement1',
+  'polylineElement2',
+  'ellipseElement1',
+  'ellipseElement2'
+].forEach(elementId => {
+  test(function() {
+    let element = document.getElementById(elementId);
+    assert_throws_dom("InvalidStateError", function() { element.getTotalLength(); });
+  }, document.title + ', ' +
+     document.getElementById(elementId).getAttribute('description'));
```

Comment:
I don't see any reason why circle, ellipse or polyline would throw (they ought to return 0). path element, given that you need an inital M attribute for the path to be valid is the only case that you could argue that throwing might be OK.

Reply:
In this case also, We can not compute getPointAtLength(0) in Firefox(throw exception) because I guess it is also not able to compute the length.

Reply:
In this case, every elements don't have attributes so i guess that each elements have default values such as 'r' and so on. I think above circle has r as 0. Actually getTotalLength() return 0, is it right?
But if r of 'circle' is 0, is it circle[1]? and also i wonder that it is possible to compute the length.

[1] http://mathforum.org/library/drmath/view/66132.html

Reply:
A <circle> is a display: none in sub-tree  -> 'r' has an initial value of 0 -> the length can be computed to 0.
But this case is the same as a <circle> has 'r' that is negative value and with a value lower or equal to zero the circle won't be drawn at all. So i thought that it deals with the same.

Reply:
is it wrong in this case?